pallets/poa/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{decl_module, decl_storage, decl_event, decl_error, dispatch,
+                    fail, traits::Get, sp_runtime::{print, SaturatedConversion} };
+use frame_system::{self as system, ensure_root};
+use sp_std::prelude::Vec;
+use codec::Decode;
+
+extern crate alloc;
+use alloc::collections::BTreeSet;
+
+/// Pallet to add and remove validators.
+
+/// The pallet's configuration trait.
+pub trait Trait: system::Trait + pallet_session::Trait {
+    // Add other types and constants required to configure this pallet.
+
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    /// Epoch length in number of slots
+    type MinEpochLength: Get<u64>;
+
+    type MaxActiveValidators: Get<u8>;
+}
+
+// This pallet's storage items.
+decl_storage! {
+
+	trait Store for Module<T: Trait> as PoAModule {
+	    /// List of active validators. Maximum allowed are `MaxActiveValidators`
+		ActiveValidators get(fn active_validators) config(): Vec<T::AccountId>;
```

Comment:
Its hard for me to read this module due to a mixing of tabs and spaces. Could you do a `find_and_replace("\t", "    ")`.